<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>820</width>  <!-- чуть расширил окно, чтобы всё влезло красиво -->
    <height>620</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Bhyve VM Manager (авторестарт)</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources.qrc">
    <normaloff>:/icons/icons/bhyve.png</normaloff>:/icons/icons/bhyve.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Параметры виртуальной машины</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <!-- Твои поля остаются без изменений -->
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Имя VM:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="lineEdit_2">
         <property name="placeholderText">
          <string>mywindows</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Память:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="lineEdit">
         <property name="placeholderText">
          <string>8G</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Диск:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="lineEdit_3">
         <property name="placeholderText">
          <string>/vm/windows.img</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>ISO (опционально):</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLineEdit" name="lineEdit_4"/>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Сеть (tap):</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QLineEdit" name="lineEdit_5">
         <property name="text">
          <string>tap0</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>

    <!-- ============ КНОПКИ УПРАВЛЕНИЯ ============ -->
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="pushButton_start">
        <property name="minimumSize">
         <size>
          <width>150</width>
          <height>40</height>
         </size>
        </property>
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: #4CAF50; color: white;</string>
        </property>
        <property name="text">
         <string>Start VM</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_stop">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>150</width>
          <height>40</height>
         </size>
        </property>
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: #f44336; color: white;</string>
        </property>
        <property name="text">
         <string>Stop VM</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_cleanupTap">
        <property name="minimumSize">
         <size>
          <width>180</width>
          <height>40</height>
         </size>
        </property>
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: #FF9800; color: white;</string>
        </property>
        <property name="text">
         <string>Force cleanup tap</string>
        </property>
        <property name="toolTip">
         <string>Уничтожить все занятые tap-интерфейсы (решает "Device busy")</string>
        </property>
       </widget>
      </item>

      <!-- === НОВАЯ КНОПКА: Скан сети === -->
      <item>
       <widget class="QPushButton" name="pushButton_arpScan">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>40</height>
         </size>
        </property>
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: #9C27B0; color: white;</string>
        </property>
        <property name="text">
         <string>Скан сети (arp-scan)</string>
        </property>
        <property name="toolTip">
         <string>Быстрое сканирование локальной сети через arp-scan --localnet</string>
        </property>
       </widget>
      </item>

      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>

    <item>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Логи</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QTextEdit" name="textEdit">
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_clear">
         <property name="text">
          <string>Очистить лог</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
